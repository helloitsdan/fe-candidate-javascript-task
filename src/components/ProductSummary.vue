<template>
  <div>
    <h3>{{ type }} product</h3>

    <p class="is-marginless">
      <strong>{{ price.monthly | formatCurrency }}</strong>
      monthly /
      <strong>{{ price.annual | formatCurrency }}</strong>
      annualy
    </p>
  </div>
</template>

<script lang="ts">
import Vue from "vue";

export default Vue.extend({
  name: "ProductSummary",
  props: {
    type: String,
    price: Object
  }
});
</script>
