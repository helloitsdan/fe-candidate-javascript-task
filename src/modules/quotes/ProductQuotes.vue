<template>
  <div class="text-center">
    <h2>Quoted Products</h2>

    <div v-if="!products">
      There are no associated products.
    </div>

    <div v-if="products" class="columns is-multiline">
      <div
        v-for="product in products"
        v-bind:key="product.type"
        class="column is-one-third"
      >
        <div class="card background--white">
          <product-summary v-bind="product" />
        </div>
      </div>
    </div>
  </div>
</template>

<script lang="ts">
import Vue from "vue";
import ProductSummary from "@/components/ProductSummary.vue";

export default Vue.extend({
  name: "ProductQuotes",
  components: {
    ProductSummary
  },
  props: {
    products: Array
  }
});
</script>
