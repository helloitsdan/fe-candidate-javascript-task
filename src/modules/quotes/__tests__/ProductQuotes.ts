import Vue from "vue";
import { shallowMount } from "@vue/test-utils";

import ProductQuotes from "../ProductQuotes.vue";
import ProductSummary from "@/components/ProductSummary.vue";

describe("ProductQuotes", () => {
  const fakeProducts = {
    products: [
      {
        type: "BASIC",
        price: {
          monthly: 100.0,
          annual: 1200.0
        }
      },
      {
        type: "SUPERIOR",
        price: {
          monthly: 200.0,
          annual: 2400.0
        }
      }
    ]
  };

  it("renders", async () => {
    const wrapper = shallowMount(ProductQuotes, {
      propsData: fakeProducts
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it("renders a ProductSummary for every customer", async () => {
    const wrapper = shallowMount(ProductQuotes, {
      propsData: fakeProducts
    });

    expect(wrapper.findAll(ProductSummary).length).toEqual(
      fakeProducts.products.length
    );
  });
});
