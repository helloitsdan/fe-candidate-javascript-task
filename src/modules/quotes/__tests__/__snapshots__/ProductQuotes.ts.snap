// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductQuotes renders 1`] = `
<div
  class="text-center"
>
  <h2>
    Quoted Products
  </h2>
   
  <!---->
   
  <div
    class="columns is-multiline"
  >
    <div
      class="column is-one-third"
    >
      <div
        class="card background--white"
      >
        <product-summary-stub
          price="[object Object]"
          type="BASIC"
        />
      </div>
    </div>
    <div
      class="column is-one-third"
    >
      <div
        class="card background--white"
      >
        <product-summary-stub
          price="[object Object]"
          type="SUPERIOR"
        />
      </div>
    </div>
  </div>
</div>
`;
